tests/python/gaia-ui-tests/gaiatest/tests/functional/lockscreen/test_lockscreen_time_check.py

```diff
@@ -0,0 +1,84 @@
+from marionette.wait import Wait
+
+from gaiatest.apps.settings.app import Settings
+from gaiatest import GaiaTestCase
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestLockScreen(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+    def test_lockscreen_time_check(self):
+        """
+        https: // bugzilla.mozilla.org / show_bug.cgi?id = 1118054
+        """
+
+        self.settings = Settings(self.marionette)
+        self.settings.launch()
+        datetime = self.settings.open_date_and_time_settings()
+
+        # Auto time update is by default set to true, turn it off to make region change
+        datetime.toggle_automatic_time_update()
+        self.assertFalse(datetime.is_autotime_enabled, 'Autotime still enabled')
+
+        # record time and change the region
+        old_time = datetime.get_current_time_text
+        datetime.set_region('Atlantic Ocean')
+
+        # get the displayed time after the region change
+        new_time = datetime.get_current_time_text
+        self.assertNotEqual(new_time, old_time)
+
+        # lock screen and check time on the lockscreen
+        self.marionette.switch_to_frame()
+        self.device.lock()
+        lock_screen = LockScreen(self.marionette)
+        self.assertLessEqual(self.get_time_difference(new_time, lock_screen.time), 1)
+
+        # configure to set the time automatically (this will revert the timezone change), then lock screen
+        lock_screen.switch_to_frame()
+        lock_screen.unlock()
+        self.apps.switch_to_displayed_app()
+
+        # Enable the auto time update, so the regions change back and date/time is reverted back
+        datetime.toggle_automatic_time_update()
+        self.assertTrue(datetime.is_autotime_enabled, 'Autotime still disabled')
+        self.marionette.switch_to_frame()
+        self.device.lock()
+
+        # wait until device is off and turn back on to check that the time is changed
+        Wait(self.marionette, timeout=20).until(
+            lambda m: not self.device.is_screen_enabled)
+        self.device.turn_screen_on()
+
+        # Check it reverted to the correct time, and compare it with the previously shown time
+        # Allow 5 minutes difference max
+        self.assertLessEqual(self.get_time_difference(old_time, lock_screen.time), 5)
+
+    def get_time_difference(self, old_time, new_lockscreen_time):
```

Comment:
AM/PM was present before 2.0 and was removed by UX: https://bugzilla.mozilla.org/show_bug.cgi?id=950884#c33

As the lockscreen doesn't display AM/PM, I would prefer to have this simple hack in the lockscreen, rather than reimplementing something which is already handled by the Python standard lib.
``` Python
    @property
    def time(self):
        # The lockscreen doesn't display AM/PM since 2.0. It's still present behind a hidden span though.
        lockscreen_time = self.marionette.execute_script("return document.getElementById('%s').innerHTML" % self._time_locator[1])
        lockscreen_time = lockscreen_time.replace('<span>', '').replace('</span>', '')
        return time.strptime(lockscreen_time, '%H:%M %p')
```